src/Application/Messenger/Handler/StripePayoutHandler.php

```diff
@@ -153,8 +159,11 @@ public function __invoke(StripePayout $payout): void
             }
         }
 
-        if ($count > 0) {
+        // Ensure we always close a transaction if we opened one, even if no charge IDs
+        // were applicable to update.
+        if (count($chargeIds) > 0) {
             $this->entityManager->flush();
```

Comment:
Are you sure we shouldn't use a separate transaction for each donation? Otherwise for any one donation will rollback all of them. Possibly we should flush each donation into the local db before we call `\MatchBot\Domain\SalesforceWriteProxyRepository::push` ?

Reply:
I'm kind of in 2 minds here @bdsl. I think there is an argument for persisting nothing if a payout contains mystery donations, but it's true that we shouldn't both push to SF and rollback. I'll try to do it with single donation txns for now.